assignment/impl/src/java/org/sakaiproject/assignment/impl/persistence/AssignmentRepositoryImpl.java

```diff
@@ -197,6 +199,26 @@ public long countAssignmentSubmissions(String assignmentId, Boolean graded, Bool
             criteria.add(Restrictions.eq("userSubmission", userSubmission));
         }
 
+        if(numSubLists > 0){
```

Comment:
There is a better solution using: org.sakaiproject.hibernate.HibernateCriterionUtils